internal/api/token_refresh.go

```diff
@@ -47,13 +47,13 @@ func (a *API) RefreshTokenGrant(ctx context.Context, w http.ResponseWriter, r *h
 		user, token, session, err := models.FindUserWithRefreshToken(db, params.RefreshToken, false)
 		if err != nil {
 			if models.IsNotFoundError(err) {
-				return oauthError("invalid_grant", "Invalid Refresh Token: Refresh Token Not Found")
+				return badRequestError(ErrorCodeRefreshTokenNotFound, "Invalid Refresh Token: Refresh Token Not Found")
 			}
 			return internalServerError(err.Error())
 		}
 
 		if user.IsBanned() {
-			return oauthError("invalid_grant", "Invalid Refresh Token: User Banned")
+			return badRequestError(ErrorCodeUserBanned, "Invalid Refresh Token: User Banned")
```

Comment:
IIRC there was some problem why I didn't add error codes with this. Can you check how backward compatible this would be?

Reply:
hmm i did a similar change for the password login flow previously - https://github.com/supabase/auth/pull/1721

but will check that it works with the client lib just to be sure that nothing breaks

Reply:
client lib handles parsing the `msg` or `error_description` fields to return any error message - https://github.com/supabase/auth-js/blob/4f21f9324b2c3d55630b8d0a6759a264b0472dd8/src/lib/fetch.ts#L35-L36